src/apple/macos/component/arm.rs

```diff
@@ -43,11 +43,17 @@ impl Components {
                 None => return,
             };
 
-            // FIXME: Do we really need to re-create a new `client` every time?
-            let client = match CFReleaser::new(IOHIDEventSystemClientCreate(kCFAllocatorDefault)) {
-                Some(c) => c,
-                None => return,
-            };
+            if self.client.is_none() {
+                let client =
+                    match CFReleaser::new(IOHIDEventSystemClientCreate(kCFAllocatorDefault)) {
+                        Some(c) => c,
+                        None => return,
+                    };
+                CFRetain(client.inner() as _);
```

Comment:
I think the original issue was because of `self.client.replace(client);`. Did you test without this call to `CFRetain`? If not, can you quickly check it please?

Reply:
yes, at first I think it was because of `self.client.replace(client)` too, and I did the test, it's not.
and from [this crash report](https://github.com/GuillaumeGomez/sysinfo/issues/816#issuecomment-1224975014)
it seems it is the dispatch_worker trying to release the client

Reply:
How weird... Oh well, that answers my question at least. Can you add a comment above the call to `CFRetain` saying something like this:

> // Without this call, `client` is freed during the execution of the program. It must be kept!